scripts/src/test/org/oppia/android/scripts/XMLSyntaxCheckTest.kt

```diff
@@ -0,0 +1,155 @@
+class XMLSyntaxCheckTest {
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = java.lang.System.out
+
+  @Rule
+  @JvmField
+  public var tempFolder: TemporaryFolder = TemporaryFolder()
+
+  @Before
+  fun setUpTests() {
+    tempFolder.newFolder("testfiles")
+    java.lang.System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    java.lang.System.setOut(originalOut)
+  }
+
+  @Test
+  fun testXmlSyntax_validXML_xmlSyntaxIsCorrect() {
+    val validXML =
+      "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n" +
+        "<shape xmlns:android=\"http://schemas.android.com/apk/res/android\"\n" +
+        "  android:shape=\"rectangle\">\n" +
+        "  <solid android:color=\"#3333334D\" />\n" +
+        "  <size android:height=\"1dp\" />\n" +
+        "</shape>"
+
+    val tempFile = tempFolder.newFile("testfiles/TestFile.xml")
+    tempFile.writeText(validXML)
+
+    runScript()
+
+    assertThat(outContent.toString().trim()).isEqualTo(
+      ScriptResultConstants.XML_SYNTAX_CHECK_PASSED
+    )
+  }
+
+  @Test
+  fun testXmlSyntax_invalidOpeningTag_xmlSyntaxIsInCorrect() {
+    val invalidXML =
+      "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n" +
+        "<shape xmlns:android=\"http://schemas.android.com/apk/res/android\"\n" +
+        "  android:shape=\"rectangle\">\n" +
+        "  <<solid android:color=\"#3333334D\" />\n" +
+        "  <size android:height=\"1dp\" />\n" +
+        "</shape>"
+    val tempFile = tempFolder.newFile("testfiles/TestFile.xml")
+    tempFile.writeText(invalidXML)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(
+      ScriptResultConstants.XML_SYNTAX_CHECK_FAILED
+    )
+    assertThat(outContent.toString().trim()).isEqualTo(
+      "XML syntax error: The content of elements must consist of well-formed" +
+        " character data or markup.\n" +
+        "lineNumber: 4\n" +
+        "columnNumber: 4\n" +
+        "File: [ROOT]/testfiles/TestFile.xml"
+    )
+  }
+
+  @Test
+  fun testXmlSyntax_wrongClosingTag_xmlSyntaxIsInCorrect() {
+    val invalidXML =
+      "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n" +
+        "<shape xmlns:android=\"http://schemas.android.com/apk/res/android\"\n" +
+        "  android:shape=\"rectangle\">\n" +
+        "  <solid android:color=\"#3333334D\" />\n" +
+        "  <size android:height=\"1dp\" />\n" +
+        "</shapes>"
+    val tempFile = tempFolder.newFile("testfiles/TestFile.xml")
+    tempFile.writeText(invalidXML)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(
+      ScriptResultConstants.XML_SYNTAX_CHECK_FAILED
+    )
+    assertThat(outContent.toString().trim()).isEqualTo(
+      "XML syntax error: The end-tag for element type \"shape\" must end" +
+        " with a '>' delimiter.\n" +
```

Comment:
This error actually seems wrong--shouldn't it be mentioning that the closing tag is missing or wrong rather than missing a ``>``?

Reply:
Yes, this might seem a bit incorrect, but the same error message can be reproduced on several of the online XML validators too. When we do the same syntax error, all of them report the same error message: 
"The end-tag for element type "shape" must end with a '>' delimiter."
https://www.xmlvalidation.com/index.php?id=1&L=0
https://www.liquid-technologies.com/online-xml-validator
https://www.truugo.com/xml_validator/
https://onlinexmltools.com/validate-xml

I think it might be fine to go with this.

Reply:
> Yes, this might seem a bit incorrect, but the same error message can be reproduced on several of the online XML validators too. When we do the same syntax error, all of them report the same error message:
> "The end-tag for element type "shape" must end with a '>' delimiter."
> https://www.xmlvalidation.com/index.php?id=1&L=0
> https://www.liquid-technologies.com/online-xml-validator
> https://www.truugo.com/xml_validator/
> https://onlinexmltools.com/validate-xml
> 
> I think it might be fine to go with this.

Reply:
Yeah, I don't think we have any choice here. I just wanted to make sure that this is working as intended, even if it isn't ideal. Thanks for clarifying.

Reply:
I think we can resolve this now.